lib/sidekiq/grouping/actor.rb

```diff
@@ -9,23 +9,53 @@ def initialize
       end
 
       private
-      
+
       def start_polling
+        return unless Sidekiq::Grouping::Config.enabled
         interval = Sidekiq::Grouping::Config.poll_interval
         info "Start polling of queue batches every #{interval} seconds"
         every(interval) { flush_batches }
       end
 
       def flush_batches
-        batches = []
+        methods = {}
 
         Sidekiq::Grouping::Batch.all.map do |batch|
-          if batch.could_flush?
-            batches << batch
+          next unless batch.could_flush?
+
+          method_name = batch.worker_class.to_s
+
+          unless methods[method_name]
+            methods[method_name] = {
+              batches: [],
+              records: 0
+            }
           end
+
+          methods[method_name][:batches] << batch
+          methods[method_name][:records] += batch.size
         end
 
-        flush(batches)
+        methods.each do |method, options|
+          job_class = method.classify.constantize
+          max_records_per_call = job_class.get_sidekiq_options["max_records_per_call"] || Sidekiq::Grouping::Config.max_records_per_call
+          max_calls_per_minute = job_class.get_sidekiq_options["max_calls_per_minute"] || Sidekiq::Grouping::Config.max_calls_per_minute
+
+          number_of_calls = options[:records] / max_records_per_call
+          number_of_calls += 1 if options[:records] % max_records_per_call > 0
```

Comment:
Nice use of mod here to determine if record / max_records has a remainder and needs to be called again to finish off the remainder batch.